### DISCUSSION TITLE
How to enable a Feature Gate

### DISCUSSION THREAD

**Discussion by RoeiDimi** [open-telemetry/opentelemetry-collector-contrib#32151] [Q&A]
Hi all,
I added a parser with a feature gate that can be seen here - https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/63eb22bf1f66d1448078bd0a85e94f6f84ac6b86/pkg/stanza/operator/parser/jsonarray/json_array_parser.go#L31

But now, i'm not sure how to enable it..
I saw [this link](https://github.com/open-telemetry/opentelemetry-collector/blob/main/featuregate/README.md) and tried running the collector with all the following variations but none worked -

otelcol --config=config.yaml --feature-gates=logs.jsonParserArray
otelcol --config=config.yaml --feature-gates=+jsonParserArray
otelcol --config=config.yaml --feature-gates=jsonParserArray

Any idea what im doing wrong?
Thanks!

**Comment by RoeiDimi** [comment]
Hi @djaglowski ,
Any chance you can help with that question?

**Reply by djaglowski** [reply]
Looks like we have a bug which causes this feature gate to be registered after we check for it: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/32313

**Reply by RoeiDimi** [reply]
@djaglowski 
would you like me to remove the flag? or move it somewhere else (not in the parser's init maybe?)

**Reply by djaglowski** [reply]
I think we need to keep the gate but issues are open already which will hopefully resolve the problem.
- https://github.com/open-telemetry/opentelemetry-collector/pull/9964
- https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32376

**Reply by ahern-pro** [reply]
I submitted https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32501 To solve this problem.

**Comment by ahern-pro** [comment]
> Hi all, I added a parser with a feature gate that can be seen here -
> 
> https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/63eb22bf1f66d1448078bd0a85e94f6f84ac6b86/pkg/stanza/operator/parser/jsonarray/json_array_parser.go#L31
> 
> But now, i'm not sure how to enable it.. I saw [this link](https://github.com/open-telemetry/opentelemetry-collector/blob/main/featuregate/README.md) and tried running the collector with all the following variations but none worked -
> 
> otelcol --config=config.yaml --feature-gates=logs.jsonParserArray otelcol --config=config.yaml --feature-gates=+jsonParserArray otelcol --config=config.yaml --feature-gates=jsonParserArray
> 
> Any idea what im doing wrong? Thanks!

I have the same problem, recommended document: `otelcol --config=config.yaml --feature-gates=logs.jsonParserArray`
